app/classifier/tasks/dropdown/index.jsx

```diff
@@ -0,0 +1,303 @@
+    // then because Province is conditional to Country it will call itself (getOptionsKey(1, 'ON'),
+    // and on second call build and return an optionsKey of 'Canada-value;ON'.
+    // this optionsKey in the City select (select.options[optionsKey]) will then provide a list cities in Ontario.
+
+    if (this.props.annotation.value[parentIndex] && this.props.annotation.value[parentIndex].option) {
+      optionsKey = this.props.annotation.value[parentIndex].value;
+      if (key) {
+        optionsKey = optionsKey.concat(`;${key}`);
+      }
+    }
+
+    if (parentSelect.condition) {
+      return this.getOptionsKey(parentIndex, optionsKey);
+    }
+    return optionsKey;
+  }
+
+  getOptions(i) {
+    const select = this.props.task.selects[i];
+    // the root select's optionsKey is '*' by default and cannot be changed, while any other select's optionsKey is based on conditional answers selected
+    const optionsKey = select.condition ? this.getOptionsKey(i) : '*';
+    const options = select.options[optionsKey];
+    return options || [];
+  }
+
+  getDisabledAttribute(i) {
+    const { selects } = this.props.task;
+    const select = selects[i];
+    const [condition] = selects.filter(filterSelect => filterSelect.id === select.condition);
+    const conditionIndex = selects.indexOf(condition);
+
+    // return of true will disable select, false will enable select
+    if (
+      select.condition && // will return false (enable select) for root/first select, which should never be disabled
+      !select.allowCreate && // will return false (enable select) if select allows create, a select with allow create should never be disabled, regagrdless of condition
+      (!this.props.annotation.value[conditionIndex] || !this.props.annotation.value[conditionIndex].option)
+      // will return true (disable select) if no conditional answer provided or the conditional answer provided is a custom/created answer (thereby making current select's options irrelevant)
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  handleFocus() {
+    this.menus[0].focus();
+  }
+
+  clearRelated(i) {
+    const { selects } = this.props.task;
+    const { id } = selects[i];
+    const relatedSelects = Object.keys(selects).filter(key => (
+      selects[key].condition === id
+    ));
+    relatedSelects.map(key => (
+      this.onChangeSelect(key, { label: null, value: null })
+    ));
+  }
+
+  selectedOptions() {
+    // return annotation values mapped to react-select option objects
+    const selectedOptions = [];
+    this.props.annotation.value.map((annotation, i) => {
+      if (annotation.option) {
+        const [selected] = this.getOptions(i).filter(option => option.value === annotation.value);
+        selectedOptions[i] = selected;
+      } else {
+        selectedOptions[i] = { label: annotation.value, value: annotation.value };
+      }
+    });
+    return selectedOptions;
```

Comment:
It just occurred to me that this could be written as an actual array map. Something like:
```javascript
const selectedOptions = this.props.annotations.map((annotation, i) => {
  if (annotation.option) {
    const [selected] = this.getOptions(i).filter(option => option.value === annotation.value);
    return selected;
  } else {
    return { label: annotation.value, value: annotation.value };
  }
});
return selectedOptions;
```